API design: Install Apple App Store apps on macOS (#19291)
fleetdm/fleet: docs/REST API/rest-api.md

    {
      "id": 121,
      "name": "Google Chrome.app",
      "available_for_install": true,
      "software_package": {

>         "name": "GoogleChrome.pkg"

Author: Add version since we have it on My device

>         "name": "GoogleChrome.pkg"

Author: ```suggestion
        "name": "GoogleChrome.pkg",
        "version": "125.12.0.3"
```

    {
      "id": 121,
      "name": "Google Chrome.app",
      "available_for_install": true,
      "software_package": {
        "name": "GoogleChrome.pkg",
        "version": "125.12.0.3"
      },

>       "app_store_app": null

Author: @mna We want to update this endpoint (`GET /hosts/:id/software`) to have similar behavior as GET /software/titles/:id - so adding available_for_install boolean + software_package for packages and app_store_app for VPP apps. One of these is always null.

    {
      "id": 121,
      "name": "Google Chrome.app",
      "software_package": {
        "name": "GoogleChrome.pkg",
        "version": "125.12.0.3",
        "self_service": true
      },

>       "app_store_app": null

Author: Hey @mna, sorry for change again. I had chat with @noahtalerman and we were thinking how could we make Software related API endpoints as consistent as possible. We think it's better to do those changes now since we just released software features and it's easier to break API.

Change: remove `available_for_install` and moving `self_service` flag into `software_package` or `app_store_app` object. This way it will work the same way as `GET /software/titles/:id` endpoint. Available for install is implied if one of the objects is not null (`software_package` or `app_store_app`)

We want same change for device's software `GET /device/:token/software` endpoint. Will follow up with PR change.

Reviewer: @marko-lisica @jahzielv just to clarify, because it becomes increasingly hard to keep track of the changes in the PR and across comments and discussions, so to recap, from the current state of the code, my understanding is that we want:

* `GET /software/titles/:id` : get a specific software title: keep it as it currently is, it already has `software_package` and `app_store_app` objects for the specific information of those parts, and it doesn't have any misplaced top-level field AFAICS.
* `GET /hosts/:id/software` : get host's software, remove `available_for_install`, move `self_service` inside `software_package` and `app_store_app`. Those sub-objects were already added as a recent change (with just the `name` and `version` for package, `app_store_id` and `version` for VPP app).
* `GET /device/:token/software`: get my device's software: exactly the same changes as for get host's software.
* `GET /software/titles`: list software titles, this is the biggest change, introduce the `app_store_app` and `software_package` sub-objects, remove `available_for_install`, move `self_service` under the sub-objects, move `icon_url` under `app_store_app`.

Does that capture the requested changes?

Author: @mna Exactly, thanks for the summary. I added a link to your comment into the subtask that I created for these changes [here](https://github.com/fleetdm/fleet/issues/20536).

> GET /device/:token/software: get my device's software: exactly the same changes as for get host's software.

For this one we want to have same response as for `GET /hosts/:id/software` as we discussed yesterday.

Reviewer 2: Thanks for the summary @mna ! Thanks for capturing it in a ticket @marko-lisica !

Reviewer 3: @marko-lisica @jahzielv  if we are removing `available_for_install` are we meant to render the available for install icon based on the presence of `software_package` or `app_store_app`? 
![image](https://github.com/user-attachments/assets/61af8e4b-51b4-4163-9800-0abc0add7783)

Reviewer 2: @ghernandez345 that sounds right to me!

}
```

Get Volume Purchasing Program (VPP)

> _Available in Fleet Premium_

Reviewer 4: ```suggestion
> This **endpoint is experimental** and may change. You can find the upcoming breaking changes [here](https://github.com/fleetdm/fleet/pull/21043/files#diff-7246bc304b15c8865ed8eaa205e9c244d0a0314e4bae60cf553dc06147c38b64R6423-R6446.

_Available in Fleet Premium_
```

Reviewer 4: Add experimental call out.

`Status: 200`

### Download software available for install

_Available in Fleet Premium._

> `GET /api/v1/fleet/software/titles/:software_title_id?alt=media`

Author: ```suggestion
`GET /api/v1/fleet/software/titles/:software_title_id/package?alt=media`
```

#### Example

> `GET /api/v1/fleet/software/titles/123?alt=media?team_id=2`

Author: ```suggestion
`GET /api/v1/fleet/software/titles/123/package?alt=media?team_id=2`
```

Body: <blob>
```

### Delete software available for install

_Available in Fleet Premium._

Deletes software that's available for install (package or App Store app).

> `DELETE /api/v1/fleet/software/titles/:software_title_id`

Author: Rename DELETE /api/v1/fleet/software/titles/:software_title_id/package. Use one API endpoint to remove uploaded package and App Store app.

Reviewer 5: This doesn't follow REST principles. since the `DELETE` here is describing the title this endpoint would logically delete the entire title and not just the installer/app associated to it.

maybe
```suggestion
`DELETE /api/v1/fleet/software/titles/:software_title_id/application`
```
could be a more neutral identifier to `package or app`

> `DELETE /api/v1/fleet/software/titles/:software_title_id`

Author: ```suggestion
`DELETE /api/v1/fleet/software/titles/:software_title_id/available_for_install`
```

> `DELETE /api/v1/fleet/software/titles/:software_title_id/available_for_install`

Author: Rename `DELETE /api/v1/fleet/software/titles/:software_title_id/package`. Use one API endpoint to remove uploaded package and App Store app.

  "software_title": {
    "id": 12,
    "name": "Firefox.app",

>     "bundle_identifier": "org.mozilla.firefox",

Author: `bundle_identifier` should be added as part of #19144

    "software_package": {

>       "type": "installer",

Author: ```suggestion
```

}
```

### Get VPP software

Returns the list of software purchased in Apple Business Manager.

`GET /api/v1/fleet/software/vpp/:team_id`

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

`GET /api/v1/fleet/software/vpp/3`

##### Default response

`Status: 200`

```json
{
  "vpp_software": {
    {
      "name": "Xcode",
      "latest_version": "15.4",
      "app_store_id": 497799835
    },
    {
      "name": "Logic Pro",
      "latest_version": "2.04",
      "app_store_id": 634148309
    },
}
}
```

### Add VPP software

_Available in Fleet Premium._

Add a VPP software form App Store to the Fleet software title.

> `POST /api/v1/fleet/software/vpp/:id`

Author: ```suggestion
`POST /api/v1/fleet/software/vpp/:app_store_id`
```

#### Parameters

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |

> | id   | integer | path | **Required.** The VPP software's ID. |

Author: ```suggestion
| app_store_id   | integer | path | **Required.** The ID of Apple App Store app. |
```

}
```

### Get VPP software

Returns the list of software purchased in Apple Business Manager.

`GET /api/v1/fleet/software/vpp/:team_id`

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

`GET /api/v1/fleet/software/vpp/3`

##### Default response

`Status: 200`

```json
{
  "vpp_software": {
    {
      "name": "Xcode",
      "latest_version": "15.4",

>       "store_id": 497799835

Author: ```suggestion
      "app_store_id": 497799835
```

    },
    {

>       "id": 2,

Author: ```suggestion
```

      "name": "Logic Pro",
      "latest_version": "2.04",

>       "store_id": 634148309

Author: ```suggestion
      "app_store_id": 634148309
```

}
```

### Get VPP software

Returns the list of software purchased in Apple Business Manager.

`GET /api/v1/fleet/software/vpp/:team_id`

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

`GET /api/v1/fleet/software/vpp/3`

##### Default response

`Status: 200`

```json
{
  "vpp_software": {
    {

>       "id": 1,

Author: ```suggestion
```

}
```

### Get available App Store apps

Returns the list of App Store (VPP) apps purchased in Apple Business Manager.

`GET /api/v1/fleet/software/app_store/:team_id`

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |

> | team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

Reviewer 5: How would the team id change the response? Wouldn't this just be the full list every time?

Or are you looking for a status in the response on if this app is already assigned to this team?

Reviewer 5: Added a suggestion below to show what I think this means (to enable the checkboxes in the UI when loading the list)

Author: I was thinking of filtering a response depending on if the app is already added to a team.

Author: I think that makes sense. Also, it's weird that we don't list apps that are in ABM in the API.

Reviewer 5: Apps that get added should definitely create a title if their isn't a matching one already so they should be. Or I might have misunderstood

Author: Sorry if I confused you. Right now idea is that this endpoint only lists VPP apps from ABM that aren't already added to a team (aren't assigned to a title). If we change this and add `added` flag, I think it's better for API users because we'll list all apps that are available in ABM, not only ones that aren't assigned to a titles.

#### Example

`GET /api/v1/fleet/software/app_store/3`

##### Default response

`Status: 200`

```json

> {

Reviewer 5: Needs pagination meta and default page size 50 I think.

Author: We discussed this with @rachaelshaw and decided to don't add pagination to this one. We're pretty sure it's a small list of apps. So far we didn't see more than 10.

Reviewer 5: The time to develop with pagination vs the time to develop without pagination is not even an hour's difference. All list endpoints should start with pagination and if you just want it all bump the `per_page` to 1000. That puts the choice to make those decisions into the api user / end user.

As an engineering organization it is irresponsible of us to make list endpoints without pagination no matter how small.

Reviewer 4: To expand on wireframe-first: we decided to not have pagination in the UI so we don't need it in the API.

We could be wrong and end up adding it later. If we need to add it quickly we can file a `P2` story.

Reviewer 5: While the urge to move fast is understandable, skipping pagination is a short-sighted approach that can lead to significant challenges and costs down the line. The minimal initial effort required to implement pagination is far outweighed by the long-term benefits of scalability, performance, user satisfaction, and cost efficiency. By incorporating pagination from the start, we future-proof the API, avoid technical debt, and position Fleet for sustainable growth and success.

I defer to the Release DRI @lukeheath but I want everyone to be aware that I will raise this every time we receive a list api (no matter how small) that does not include pagination by default.

Reviewer 4: Thanks George. It's not a speed thing. Wireframe/UI-first the way we approach UI/API design at Fleet.

I think it's up to @rachaelshaw (API design DRI).

Reviewer 6: Yeah worth keeping in mind that anything we paginate by convention has the "< Previous" + "Next >" buttons in the UI; if those are always expected to be disabled, or feel out-of-place, then probably doesn't need pagination.

(Also, can always add later if needed.)

Reviewer 5: we can still build a UI that calls /endpoint&page_size=1000 and not develop UI pagination atm but in the backend it should always have pagination to start

  "app_store_apps": {

>     {
>       "name": "Xcode",
>       "latest_version": "15.4",
>       "app_store_id": 497799835
>     },
>     {
>       "name": "Logic Pro",
>       "latest_version": "2.04",
>       "app_store_id": 634148309
>     },

Reviewer 5: ```suggestion
    {
      "name": "Xcode",
      "latest_version": "15.4",
      "app_store_id": 497799835,
      "added": true
    },
    {
      "name": "Logic Pro",
      "latest_version": "2.04",
      "app_store_id": 634148309,
      "added": false
    },
```

Reviewer 5: Also I think this API call is missing the `pngs` you expect in the response to be able to show the icons in the UI list.

Author: > Also I think this API call is missing the `pngs` you expect in the response to be able to show the icons in the UI list.

I missed this, thanks

}
}
```

### Add App Store app

_Available in Fleet Premium._

Add App Store (VPP) app purchased in Apple Business Manager.

`POST /api/v1/fleet/software/app_store/:app_store_id`

#### Parameters

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |
| app_store_id   | integer | path | **Required.** The ID of App Store app. |

> | team_id       | integer | query | **Required**. The team ID. Adds VPP software to the specified team.  |

Author: Do you think if we go this way it would make it harder to support multiple apps later? We can just document the new `app_stor_ids` param?

> | team_id       | integer | query | **Required**. The team ID. Adds VPP software to the specified team.  |

Reviewer 5: It wouldn't be harder in the future it would just be on a different team endpoint

## Software

> - [Add software package](#add-software)

Reviewer 4: Hey @marko-lisica do we need to add "package" to the name's here? I think it's simpler to have "Add software", "Download software", etc.

Author: Add software feature in the UI has 2 options for now - packages and App Store apps (VPP). Packages and App Store apps are subsets of **Add software** feature. Soon we'll have Fleet library which is also the type of software available for install. I think being specific with this one makes sense. So we would have `Add software package`, `Add App Store app` and `Add Fleet library app`. 
 
Download endpoint is currently used with package, but could be used with Fleet library apps as well. Delete endpoint is used with all types of software available for install. Based on this we should probably name it `Delete software available for install` and `Download software available for install`.

Wdyt?

Reviewer 4: > Packages and App Store apps are subsets of Add software feature. Soon we'll have Fleet library which is also the type of software available for install. I think being specific with this one makes sense. So we would have Add software package, Add App Store app and Add Fleet library app.

Ah, I forgot we have two API endpoints:
- `POST /software/package`
- `POST /software/app_store_app`

We can use `POST /software/package` for Fleet library right?

Maybe we just call this "Add package" and "Add App Store app." Because we're already under "Software" and it's less words.

And then maybe we go w/ "Delete package or App Store app" and "Download package" to be more explicit (also less words)

IIRC correctly, we don't have a feature to download an App Store app.

Thoughts?

Author: > We can use POST /software/package for Fleet library right?

@noahtalerman I'm not sure since it is `multipart/form-data` request. For the library apps Fleet will import files into a title. But no matter if we have a new endpoint or use this one I think we can call it `Add package`

> - [Download software package](#download-software)

Author: ```suggestion
- [Download software available for install](#download-software-available-for-install)
```

> - [Delete software available for install](#delete-software)

Author: ```suggestion
- [Delete software available for install](#delete-software-available-for-install)
```

Body: <blob>
```

### Delete software available for install

_Available in Fleet Premium._

Deletes software that's available for install (package or App Store app).

`DELETE /api/v1/fleet/software/titles/:software_title_id/available_for_install`

#### Parameters

| Name            | Type    | In   | Description                                      |
| ----            | ------- | ---- | --------------------------------------------     |
| software_title_id              | integer | path | **Required**. The ID of the software title to delete software available for install. |
| team_id | integer | query | **Required**. The team ID. Deletes a software package added to the specified team. |

#### Example

> `DELETE /api/v1/fleet/software/titles/24?team_id=2`

Author: ```suggestion
`DELETE /api/v1/fleet/software/titles/24/available_for_install?team_id=2`
```

    {
      "id": 12,
      "name": "Firefox.app",

>       "available_for_install": true,

Author: ```suggestion
      "available_for_install": true,
      "icon_url": null,
```

    {
      "id": 12,
      "name": "Firefox.app",
      "software_package": {
        "name": "FirefoxInsall.pkg",
        "version": "125.6",

>         "icon_url": null,

Author: ```suggestion
```

    {
      "id": 22,
      "name": "Google Chrome.app",
      "available_for_install": false,

>       "app_store_app": null,

Author: ```suggestion
   
```

    {
      "id": 32,
      "name": "1Password – Password Manager",

>       "available_for_install": false,

Author: #### Dev note:
We think that breaking change is ok here. It gives us more flexibility for the future if we want to bring `software_package` as object with information that we have in `GET software/titles/:id`

@rachaelshaw Wdyt?

Reviewer: Sounds good, thanks.

  "software_title": {
    "id": 12,
    "name": "Firefox.app",
    "bundle_identifier": "org.mozilla.firefox",

>     "software_package": {

Author: #### Dev note:

For VPP apps `software_package` will be `null` and new object will be there instead (`app_store_app`). See in [Figma](https://www.figma.com/design/QsO3Q03QGVbiSknGpLqgAB/%2318867-Install-Apple-App-Store-apps-on-macOS?node-id=494-9763&t=QY0QlnuIig1o8gZN-11).

}
```

#### Example (App Store app)

`GET /api/v1/fleet/software/titles/15`

##### Default response

`Status: 200`

```json
{
  "software_title": {
    "id": 15,
    "name": "Logic Pro",
    "bundle_identifier": "com.apple.logic10",
    "software_package": null,
    "app_store_app": {
      "name": "Logic Pro",

>       "app_store_id": 1091189122,

Reviewer 7: ```suggestion
      "app_store_id": "1091189122",
```

}
```

### Get available App Store apps

Returns the list of App Store (VPP) apps purchased in Apple Business Manager.

> `GET /api/v1/fleet/software/app_store/:team_id`

Author: ```suggestion
`GET /api/v1/fleet/software/app_store_apps/:team_id`
```

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

> `GET /api/v1/fleet/software/app_store/3`

Author: ```suggestion
`GET /api/v1/fleet/software/app_store_apps/3`
```

##### Default response

`Status: 200`

```json
{
  "app_store_apps": {
    {
      "name": "Xcode",
      "latest_version": "15.4",
      "app_store_id": 497799835,
      "added": true
    },
    {
      "name": "Logic Pro",
      "latest_version": "2.04",
      "app_store_id": 634148309,
      "added": false
    },
}
}
```

### Add App Store app

_Available in Fleet Premium._

Add App Store (VPP) app purchased in Apple Business Manager.

> `POST /api/v1/fleet/software/app_store/:app_store_id`

Author: ```suggestion
`POST /api/v1/fleet/software/app_store_apps/:app_store_id`
```

#### Parameters

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |
| app_store_id   | integer | path | **Required.** The ID of App Store app. |
| team_id       | integer | query | **Required**. The team ID. Adds VPP software to the specified team.  |

#### Example

> `POST /api/v1/fleet/software/app_store/2?team_id=3`

Author: ```suggestion
`POST /api/v1/fleet/software/app_store_apps/2?team_id=3`
```

}
```

### Get available App Store apps

Returns the list of App Store (VPP) apps purchased in Apple Business Manager.

`GET /api/v1/fleet/software/app_store_apps/:team_id`

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

`GET /api/v1/fleet/software/app_store_apps/3`

##### Default response

`Status: 200`

```json
{
  "app_store_apps": {
    {

>       "name": "Xcode",

Author: ```suggestion
      "name": "Xcode",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
```

}
```

### Get available App Store apps

Returns the list of App Store (VPP) apps purchased in Apple Business Manager.

`GET /api/v1/fleet/software/app_store_apps/:team_id`

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

`GET /api/v1/fleet/software/app_store_apps/3`

##### Default response

`Status: 200`

```json
{
  "app_store_apps": {
    {
      "name": "Xcode",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "15.4",
      "app_store_id": 497799835,
      "added": true
    },
    {

>       "name": "Logic Pro",

Author: ```suggestion
      "name": "Logic Pro",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
```

}
```

### Get available App Store apps

> Returns the list of App Store (VPP) apps purchased in Apple Business Manager.

Author: ```suggestion
Returns the list of App Store (VPP) apps purchased in Apple Business Manager. Apps that are already added to a team won't be returned.```

> `GET /api/v1/fleet/software/app_store_apps/:team_id`

Reviewer 5: Changing this to query_param ex. `team_id=2`

> `GET /api/v1/fleet/software/app_store_apps/:team_id`

Author: ```suggestion
`GET /api/v1/fleet/software/app_store_apps`
```

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

> `GET /api/v1/fleet/software/app_store_apps/3`

Author: ```suggestion
`GET /api/v1/fleet/software/app_store_apps/?team_id=3`
```

##### Default response

`Status: 200`

```json
{
  "app_store_apps": {
    {
      "name": "Xcode",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "15.4",
      "app_store_id": 497799835,
      "added": true
    },
    {
      "name": "Logic Pro",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "2.04",
      "app_store_id": 634148309,

>       "added": false

Reviewer 2: @marko-lisica just wanted to call out that this field will always be false because the backend will only return apps that have not been added to this team (per the note here: https://github.com/fleetdm/fleet/issues/18867 under `Changes > Product`)

if we want to get rid of it, it's a 1 line change in the BE so no biggy :D

| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

`GET /api/v1/fleet/software/app_store_apps/?team_id=3`

##### Default response

`Status: 200`

```json
{
  "app_store_apps": {
    {
      "name": "Xcode",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "15.4",
      "app_store_id": 497799835,
      "added": true
    },
    {
      "name": "Logic Pro",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "2.04",
      "app_store_id": 634148309,
      "added": false
    },
}
}
```

### Add App Store app

_Available in Fleet Premium._

Add App Store (VPP) app purchased in Apple Business Manager.

`POST /api/v1/fleet/software/app_store_apps/:app_store_id`

#### Parameters

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |

> | app_store_id   | integer | path | **Required.** The ID of App Store app. |

Author: ```suggestion
| app_store_id   | integer | body | **Required.** The ID of App Store app. |
```

| team_id       | integer | query | **Required**. The team ID. Adds VPP software to the specified team.  |

#### Example

> `POST /api/v1/fleet/software/app_store_apps/2?team_id=3`

Author: ```suggestion
`POST /api/v1/fleet/software/app_store_apps?team_id=3`

##### Request body

```json
{
  "app_store_id": 2
}
```
```

    {
      "name": "Xcode",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "15.4",
      "app_store_id": 497799835,
      "added": true
    },
    {
      "name": "Logic Pro",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "2.04",
      "app_store_id": 634148309,
      "added": false
    },
}
}
```

### Add App Store app

_Available in Fleet Premium._

Add App Store (VPP) app purchased in Apple Business Manager.

`POST /api/v1/fleet/software/app_store_apps/:app_store_id`

#### Parameters

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |
| app_store_id   | integer | path | **Required.** The ID of App Store app. |
| team_id       | integer | query | **Required**. The team ID. Adds VPP software to the specified team.  |

#### Example

`POST /api/v1/fleet/software/app_store_apps?team_id=3`

##### Request body

```json
{
  "app_store_id": 2
}

> 

Author: ```suggestion
```
```

`Status: 200`

```json
{
  "app_store_apps": {
    {
      "name": "Xcode",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "15.4",
      "app_store_id": 497799835,
      "added": true
    },
    {
      "name": "Logic Pro",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "2.04",
      "app_store_id": 634148309,
      "added": false
    },
}
}
```

### Add App Store app

_Available in Fleet Premium._

Add App Store (VPP) app purchased in Apple Business Manager.

`POST /api/v1/fleet/software/app_store_apps`

#### Parameters

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |
| app_store_id   | integer | body | **Required.** The ID of App Store app. |
| team_id       | integer | body | **Required**. The team ID. Adds VPP software to the specified team.  |

#### Example

`POST /api/v1/fleet/software/app_store_apps?team_id=3`

##### Request body

```json
{

>   "app_store_id": 2,

Author: ```suggestion
  "app_store_id": 497799835,
```

}
```

### Get available App Store apps

Returns the list of App Store (VPP) apps purchased in Apple Business Manager. Apps that are already added to a team won't be returned.

`GET /api/v1/fleet/software/app_store_apps`

#### Parameters

| Name    | Type | In | Description |
| ------- | ---- | -- | ----------- |
| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |

#### Example

`GET /api/v1/fleet/software/app_store_apps/?team_id=3`

##### Default response

`Status: 200`

```json
{
  "app_store_apps": {
    {
      "name": "Xcode",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "15.4",

>       "app_store_id": 497799835

Reviewer 7: ```suggestion
      "app_store_id": "497799835"
```

    },
    {
      "name": "Logic Pro",
      "icon_url": "https://is1-ssl.mzstatic.com/image/thumb/Purple211/v4/f1/65/1e/a4844ccd-486d-455f-bb31-67336fe46b14/AppIcon-1x_U007emarketing-0-7-0-85-220-0.png/512x512bb.jpg",
      "latest_version": "2.04",

>       "app_store_id": 634148309

Reviewer 7: ```suggestion
      "app_store_id": "634148309"
```

    },
}
}
```

### Add App Store app

_Available in Fleet Premium._

Add App Store (VPP) app purchased in Apple Business Manager.

`POST /api/v1/fleet/software/app_store_apps`

#### Parameters

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |

> | app_store_id   | integer | body | **Required.** The ID of App Store app. |

Reviewer 7: ```suggestion
| app_store_id   | string | body | **Required.** The ID of App Store app. |
```

| team_id       | integer | body | **Required**. The team ID. Adds VPP software to the specified team.  |

#### Example

`POST /api/v1/fleet/software/app_store_apps?team_id=3`

##### Request body

```json
{

>   "app_store_id": 497799835,

Reviewer 7: ```suggestion
  "app_store_id": "497799835",
```
